File: src/main/java/com/java/date/GetDate.java
package com.java.date;

import java.util.Calendar;
import java.util.Date;

/**
 * @author zJiaLi
 * @since 2020-03-25 10:19
 */
public class GetDate {
    public static void main(String[] args) {
        GetDate getDate = new GetDate();
        Thread[] threads = new Thread[100];
        for (int i = 0; i < 100; i++) {
            Thread t1 = new Thread(() -> {

                Date startTime = null;
                for (int j = 0; j < 20; j++) {
                    startTime = getDate.getStartTime();
                }
                System.out.println(Thread.currentThread().getName());
                System.out.println(startTime);
            });
            threads[i] = t1;
        }

        for (Thread thread : threads) {
            thread.start();
        }

    }

    public Date getStartTime() {
        Calendar calendar = Calendar.getInstance();
        calendar.setTime(new Date());
        calendar.set(Calendar.HOUR_OF_DAY, 23);
        calendar.set(Calendar.MINUTE, 59);
        calendar.set(Calendar.SECOND, 59);
        return calendar.getTime();
    }
}

File: src/main/java/com/java/date/GetDate.kt
package com.java.date

import java.util.*

/**
 * @author zJiaLi
 * @since 2020-03-25 10:19
 */
class GetDate {
    val startTime: Date
        get() {
            val calendar = Calendar.getInstance()
            calendar.time = Date()
            calendar[Calendar.HOUR_OF_DAY] = 23
            calendar[Calendar.MINUTE] = 59
            calendar[Calendar.SECOND] = 59
            return calendar.time
        }

    companion object {
        @JvmStatic
        fun main(args: Array<String>) {
            val getDate = GetDate()
            val threads = arrayOfNulls<Thread>(100)
            for (i in 0..99) {
                val t1 = Thread(Runnable {
                    var startTime: Date? = null
                    for (j in 0..19) {
                        startTime = getDate.startTime
                    }
                    println(Thread.currentThread().name)
                    println(startTime)
                })
                threads[i] = t1
            }
            for (thread in threads) {
                thread!!.start()
            }
        }
    }
}